text/0019-stricter-validation.md

```diff
@@ -150,3 +136,7 @@ Mitigating this issue is that the backward incompatibility will be minimal in pr
 ## Alternatives
 
 The proposal is, we believe, the minimal change to make strict mode self-contained, but not too different from what was there before. It also should not result in pervasive changes to the existing code. Alternative designs we considered (e.g., allowing more permissive mode features in strict mode) would be more invasive.
+
+## Updates
+
+* 2023-11-07: The original text said that `AnyEntity` was used to type unspecified entities. This is not the case in any released version of Cedar -- instead unspecified entities are given a special `Unspecified` type. Updated text to reflect this.
```

Comment:
Even for validation in permissive mode ?

Reply:
Yep, both modes. The type of an unspecified entity (i.e., a principal/resource in an environment where the action was missing `appliesTo` field(s) in the schema) is `Entity({Unspecified})` rather than `AnyEntity`. `AnyEntity` is only used for template slots and the UB of actions.

In typechecking, `<Unspecified> == <Specified Entity>` is typed as `False`.

Reply:
Just double checked the code, and I was too hasty 😬 Here is the revised answer:

"Unspecified" entities are given type `AnyEntity` by the type checker, but in the rules for typing `==` and `in` are special-cased to return `False` when one element is unspecified.

Reply:
This has been true since the initial release, see [`type_of_equality`](https://github.com/cedar-policy/cedar/blob/release/2.0.x/cedar-policy-validator/src/typecheck.rs#L1916) and [`typecheck_in`](https://github.com/cedar-policy/cedar/blob/release/2.0.x/cedar-policy-validator/src/typecheck.rs#L1975).

Will update the note to accurately reflect this. Sorry for the bugs... I have a lot of balls in the air this week.

Reply:
Updated